tests/utils.ts

```diff
@@ -17,6 +17,69 @@ export enum TableTypes {
   AllTypes,
 }
 
+type TablePlotNames =
+  | 'all_types'
+  | 'simple_plot'
+  | 'simple_table'
+  | 'simple_table_header_group'
+  | 'simple_table_header_group_hide'
+  | 'double_and_string'
+  | 'ordered_int_and_offset'
+  | 'trig_table'
+  | 'trig_figure'
+  | 'multiselect_null'
+  | 'multiselect_empty'
+  | 'multiselect_bool'
+  | 'multiselect_datetime'
+  | 'multiselect_char'
+  | 'multiselect_number'
+  | 'multiselect_string';
+
+/**
+ * Opens a table loaded from application mode and returns the grid location
+ * @param page
+ * @param type Either 'table' or 'plot'
+ * @param name Name of the table or plot
+ */
+export async function openTableOrPlot(
+  page: Page,
+  type: 'table' | 'plot',
+  name: TablePlotNames
+): Promise<void> {
+  await page.goto('');
+  await expect(page.locator('.loading-spinner')).toHaveCount(0);
+
+  // open the tables/plot button
+  const dropdownButton = page.locator(
+    type === 'table' ? '#table-actions' : '#widget-actions'
+  );
+  expect(dropdownButton).not.toBeNull();
+  expect(dropdownButton).not.toBeDisabled();
+  await dropdownButton.click();
+
+  // search for the table/plot
+  const search = page.getByPlaceholder('Search');
+  expect(search).not.toBeNull();
+  expect(search).not.toBeDisabled();
+  await search.type(name);
+
+  // open the table/plot
+  const openButton = page.locator('.btn-context-menu').first();
+  expect(openButton).not.toBeNull();
+  expect(openButton).not.toBeDisabled();
+  await openButton.click();
+
+  await expect(
+    page.locator('.iris-grid .iris-grid-loading-status')
+  ).toHaveCount(0);
+
+  // get grid and return that
+  // const grid = await page.locator('.iris-grid-panel .iris-grid');
+  // const gridLocation = await grid.boundingBox();
+  // expect(gridLocation).not.toBeNull();
+  // return gridLocation;
```

Comment:
Remove unused code, and also this part about checking for `iris-grid` should really only be checked for tables loaded. I'd have a separate method `openTable` that calls this `openObject` method, e.g.

```
export async function openTable(page: Page, name: TableNames, waitForLoadFinished = true): Promise<void> {
  await openObject(page, 'table', name);

  if (waitForLoadFinished) {
    await expect(
      page.locator('.iris-grid .iris-grid-loading-status')
    ).toHaveCount(0);
  }
}
```

And a similar `openPlot` function as well.